google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -550,6 +584,13 @@ private SessionPoolTransactionRunner(SessionPool sessionPool, PooledSession sess
             result = runner.run(callable);
             break;
           } catch (SessionNotFoundException e) {
+            sessionReplacementCount++;
+            if (sessionReplacementCount > MAX_SESSION_REPLACES) {
```

Comment:
same as above.. just leave this comment open until there is a better strategy defined for prev instance